authentication.go

```diff
@@ -89,6 +106,13 @@ func (s *AuthenticationService) AcquireSessionCookie(username, password string)
 	return true, nil
 }
 
+// Sets the access token on the Authentication Service (helpful in event access tokens are cached by the client code).
+// Use GetOauth2AccessToken to get a new access token and set authType to Oauth
+func (s *AuthenticationService) SetOauth(accessToken string) {
```

Comment:
Same with version number, or pass version number to the method:

*Method name*
`SetOauth2`

*Argument*
`SetOauth(accessToken string, version int)`